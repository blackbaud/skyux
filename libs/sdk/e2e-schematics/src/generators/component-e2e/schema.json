{
  "$schema": "http://json-schema.org/schema",
  "$id": "library-e2e",
  "title": "Create a library e2e",
  "description": "Creates an e2e infrastructure for a library.",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The name of the library.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What is the name of the library?",
      "pattern": "^[a-zA-Z].*$"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the library (used for linting)."
    }
  },
  "additionalProperties": false,
  "required": ["name"]
}
