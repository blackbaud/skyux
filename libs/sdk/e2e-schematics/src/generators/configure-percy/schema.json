{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "configure-percy",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "e2e project name",
      "$default": {
        "$source": "argv",
        "index": 0
      }
    },
    "skipFormat": {
      "type": "boolean",
      "aliases": ["skip-format"],
      "description": "Skip formatting files.",
      "default": false
    }
  }
}
