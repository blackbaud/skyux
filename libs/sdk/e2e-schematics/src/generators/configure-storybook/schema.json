{
  "$schema": "http://json-schema.org/schema",
  "$id": "configure-storybook",
  "title": "Configure Storybook",
  "description": "Adds configuration to Storybook.",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The name of the project using Storybook.",
      "$default": {
        "$source": "argv",
        "index": 0
      }
    }
  },
  "additionalProperties": false
}
