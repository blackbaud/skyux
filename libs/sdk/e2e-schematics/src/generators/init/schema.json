{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "Init",
  "title": "Configure E2E environment"
}
