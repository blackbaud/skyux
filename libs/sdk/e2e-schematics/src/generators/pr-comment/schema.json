{
  "$schema": "http://json-schema.org/schema",
  "$id": "PrComment",
  "title": "PR Comment",
  "description": "Generate markdown for a PR comment or a PR preview README.md",
  "type": "object",
  "properties": {
    "pr": {
      "type": "number",
      "description": "PR Number",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "Which PR is this for?"
    },
    "storybooks": {
      "type": "string",
      "description": "Comma separated list of storybooks to include in the comment",
      "alias": "s",
      "x-prompt": "Which storybook projects should be listed?"
    },
    "apps": {
      "type": "string",
      "description": "Comma separated list of apps to include in the comment",
      "alias": "a"
    }
  },
  "required": ["pr", "storybooks"]
}
