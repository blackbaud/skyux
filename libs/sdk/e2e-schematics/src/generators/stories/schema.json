{
  "$schema": "http://json-schema.org/schema",
  "$id": "Stories",
  "title": "Create Storybook stories/specs",
  "description": "Creates Storybook stories/specs for all Angular components declared in a project.",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What's the name of the project for which you want to generate stories?",
      "aliases": [
        "app",
        "storybook-app",
        "storybookapp",
        "storybookApp",
        "storybook"
      ]
    },
    "generateCypressSpecs": {
      "type": "boolean",
      "description": "Specifies whether to automatically generate `*.spec.ts` files in the Cypress e2e app generated by the `cypress-configure` generator.",
      "default": true
    },
    "cypressProject": {
      "type": "string",
      "description": "The Cypress project to generate the stories under. This is inferred from `name` by default."
    }
  },
  "additionalProperties": false,
  "required": ["project"]
}
