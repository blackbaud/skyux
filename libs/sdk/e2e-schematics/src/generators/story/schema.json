{
  "$schema": "http://json-schema.org/schema",
  "$id": "Story",
  "title": "Create a new story with a new component and corresponding tests",
  "description": "Create a new story with a new component and corresponding tests",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "The name of the component.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the new storybook component?"
    },
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "x-prompt": "What's the name of the storybook application for which you want to generate the story?",
      "aliases": [
        "app",
        "storybook-app",
        "storybookapp",
        "storybookApp",
        "storybook"
      ]
    },
    "generateCypressSpecs": {
      "type": "boolean",
      "description": "Specifies whether to automatically generate `*.spec.ts` files in the Cypress e2e app generated by the `cypress-configure` generator.",
      "default": true
    },
    "cypressProject": {
      "type": "string",
      "description": "The Cypress project to generate the stories under. This is inferred from `project` by default."
    },
    "includeTests": {
      "type": "boolean",
      "description": "Include tests in the library.",
      "default": false
    }
  },
  "additionalProperties": false,
  "required": ["name", "project"]
}
