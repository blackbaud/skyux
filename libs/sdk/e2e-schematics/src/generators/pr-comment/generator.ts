import { Tree, generateFiles, joinPathFragments } from '@nx/devkit';

import { PrCommentGeneratorOptions, PrCommentGeneratorSchema } from './schema';

function normalizeOptions(
  tree: Tree,
  options: PrCommentGeneratorSchema
): PrCommentGeneratorOptions {
  const pr = options.pr;
  const storybooks = options.storybooks
    .split(',')
    .map((s) => s.trim())
    .filter((s) => s.length > 0);
  const apps = (options.apps || '')
    .split(',')
    .map((s) => s.trim())
    .filter((s) => s.length > 0);
  if (!pr || isNaN(parseInt(`${pr}`, 10))) {
    throw new Error('Invalid PR number');
  }
  return {
    pr,
    storybooks,
    apps,
  };
}

export default async function prComment(
  tree: Tree,
  options: PrCommentGeneratorSchema
) {
  const normalizedOptions = normalizeOptions(tree, options);
  generateFiles(
    tree,
    joinPathFragments(__dirname, 'files/comment'),
    'dist',
    normalizedOptions
  );
  generateFiles(
    tree,
    joinPathFragments(__dirname, 'files/readme'),
    'dist',
    normalizedOptions
  );
}
