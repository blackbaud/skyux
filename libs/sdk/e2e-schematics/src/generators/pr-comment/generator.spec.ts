import { Tree } from '@nx/devkit';

import { createTree } from 'nx/src/generators/testing-utils/create-tree';

import generator from './generator';
import { PrCommentGeneratorSchema } from './schema';

describe('pr-comment generator', () => {
  let appTree: Tree;
  const options: PrCommentGeneratorSchema = {
    pr: '123',
    storybooks: 'storybook-1,storybook-2',
  };

  beforeEach(() => {
    appTree = createTree();
  });

  it('should generate files', async () => {
    await generator(appTree, options);
    expect(appTree.isFile('dist/pr_comment.md')).toBeTruthy();
    expect(appTree.read('dist/pr_comment.md', 'utf-8')).toMatchSnapshot(
      'dist/pr_comment.md'
    );
    expect(appTree.isFile('dist/README.md')).toBeTruthy();
    expect(appTree.read('dist/README.md', 'utf-8')).toMatchSnapshot(
      'dist/README.md'
    );
  });

  it('should generate files with apps', async () => {
    await generator(appTree, {
      ...options,
      apps: 'app-1,app-2',
    });
    expect(appTree.isFile('dist/pr_comment.md')).toBeTruthy();
    expect(appTree.read('dist/pr_comment.md', 'utf-8')).toMatchSnapshot(
      'dist/pr_comment.md'
    );
    expect(appTree.isFile('dist/README.md')).toBeTruthy();
    expect(appTree.read('dist/README.md', 'utf-8')).toMatchSnapshot(
      'dist/README.md'
    );
  });

  it('should generate files with no storybooks', async () => {
    await generator(appTree, {
      ...options,
      storybooks: '',
      apps: 'app-1,app-2',
    });
    expect(appTree.isFile('dist/pr_comment.md')).toBeTruthy();
    expect(appTree.read('dist/pr_comment.md', 'utf-8')).toMatchSnapshot(
      'dist/pr_comment.md'
    );
    expect(appTree.isFile('dist/README.md')).toBeTruthy();
    expect(appTree.read('dist/README.md', 'utf-8')).toMatchSnapshot(
      'dist/README.md'
    );
  });

  it('should throw errors', async () => {
    await expect(
      generator(appTree, {
        ...options,
        pr: '',
      })
    ).rejects.toThrowErrorMatchingSnapshot();
  });
});
