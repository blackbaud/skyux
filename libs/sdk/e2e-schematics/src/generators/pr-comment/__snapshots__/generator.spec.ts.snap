// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pr-comment generator should generate files: dist/README.md 1`] = `
"# Sky UX Pull Request Preview

Preview build for [PR #123](https://github.com/blackbaud/skyux/pull/123)

[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/123/storybook/)

Component Storybooks:

- [storybook-1](https://blackbaud.github.io/skyux-pr-preview/123/storybooks/storybook-1/)
- [storybook-2](https://blackbaud.github.io/skyux-pr-preview/123/storybooks/storybook-2/)
"
`;

exports[`pr-comment generator should generate files: dist/pr_comment.md 1`] = `
"[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/123/storybook/)

Component Storybooks:

- [storybook-1](https://blackbaud.github.io/skyux-pr-preview/123/storybooks/storybook-1/)
- [storybook-2](https://blackbaud.github.io/skyux-pr-preview/123/storybooks/storybook-2/)
"
`;

exports[`pr-comment generator should throw errors 1`] = `"Invalid PR number"`;

exports[`pr-comment generator should throw errors 2`] = `"No storybooks specified"`;
